src/dex-ui-components/base/Inputs/FileUploader.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback } from "react";
+import { useDropzone } from "react-dropzone";
+import { Card, Text } from "@chakra-ui/react";
+import { Color } from "../../../dex-ui-components";
+interface File extends Blob {
+  readonly lastModified: number;
+  readonly name: string;
+  readonly webkitRelativePath: string;
+}
+
+const defaultFileUploaderStyles = {
+  bg: "white",
+  width: "100%",
+  textAlign: "center",
+  justifyContent: "center",
+  alignSelf: "center",
+  border: "1px #31A9BD dashed",
+  height: "102Px",
+};
+interface FileUploaderProps {
+  title: string;
+  subTitle: string;
+  onHoverTitle: string;
+  fileUploaderStyles?: any;
+  disabled?: boolean;
+  onFileRead: (file: string, fileName: string) => void;
+  onFileReadFailed?: (reason: string) => void;
+}
+
+function FileUploader(props: FileUploaderProps) {
+  const { title, subTitle, onHoverTitle, fileUploaderStyles, disabled, onFileRead, onFileReadFailed } = props;
+
+  const onDrop = useCallback(
```

Comment:
There's no need to use `useCallback` unless we're optimizing for a specific performance issue. It adds unnecessary bloat to the code and can even hurt performance (https://kentcdodds.com/blog/usememo-and-usecallback). You may have already noticed this but `useCallback` is unnecessarily being used in other components as well and will need to be refactored at a later point.